niceman/resource/tests/test_ssh.py

```diff
@@ -0,0 +1,53 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+import logging
+import re
+import six
+
+from ...utils import swallow_logs
+from ...tests.utils import assert_in, skip_if_no_network
+from ..base import ResourceManager
+
+
+@skip_if_no_network
+def test_ssh_class():
+
+    with swallow_logs(new_level=logging.DEBUG) as log:
+
+        # Test connecting to test SSH server.
+        # TODO: Add a test using a SSH key pair.
+        config = {
+            'name': 'ssh-test-resource',
+            'type': 'ssh',
+            'host': 'localhost',
+            'user': 'root',
+            'password': 'root',
+            'port': '49000'
+        }
+        resource = ResourceManager.factory(config)
+        updated_config = resource.create()
+        config.update(updated_config)
+        assert re.match('\w{8}-\w{4}-\w{4}-\w{4}-\w{12}',
```

Comment:
you might want to add a trailing anchor `$` since apparently (as I have discovered after years of blessed ignorance) `match` just guarantees matching from the beginning... e.g.
```python
$> python -c 'import re; print(re.match("\w{4}", "toolong"))'         
<_sre.SRE_Match object at 0x7f50b7980718>

$> python -c 'import re; print(re.match("\w{4}$", "toolong"))' 
None
```